src/data-type.js

```diff
@@ -1280,13 +1280,14 @@ const TYPE = module.exports.TYPE = {
 
     writeParameterData: function(buffer, parameter, options) {
       if (parameter.value != null) {
-        let ref, time = new Date(+parameter.value);
+        let time = new Date(+parameter.value);
         if (options.useUTC) {
           time = ((time.getUTCHours() * 60 + time.getUTCMinutes()) * 60 + time.getUTCSeconds()) * 1000 + time.getUTCMilliseconds();
         } else {
           time = ((time.getHours() * 60 + time.getMinutes()) * 60 + time.getSeconds()) * 1000 + time.getMilliseconds();
         }
-        time = (time / 1000 + ((ref = parameter.value.nanosecondDelta) != null ? ref : 0)) * Math.pow(10, parameter.scale);
+        time = Math.round(time * Math.pow(10, parameter.scale - 3));
+        time += Math.round((parameter.value.nanosecondDelta != null ? parameter.value.nanosecondDelta : 0) * Math.pow(10, parameter.scale));
```

Comment:
Please do the <code>Math.round()</code> after the addition. Like below:
```javascript
time = time * Math.pow(10, parameter.scale - 3);
time += (parameter.value.nanosecondDelta != null ? parameter.value.nanosecondDelta : 0) * Math.pow(10, parameter.scale);
time = Math.round(time);
```
Also, can you please add tests?

Thanks for fixing this!